controllers/flowlogspipeline/flp_objects.go

```diff
@@ -159,61 +242,56 @@ func (b *builder) podTemplate(configDigest string) corev1.PodTemplateSpec {
 					},
 				},
 			}},
-			Containers: []corev1.Container{{
-				Name:            constants.GoflowKubeName,
-				Image:           b.desired.Image,
-				ImagePullPolicy: corev1.PullPolicy(b.desired.ImagePullPolicy),
-				Command:         []string{"/bin/sh", "-c", cmd},
-				Resources:       *b.desired.Resources.DeepCopy(),
-				VolumeMounts: []corev1.VolumeMount{{
-					MountPath: configPath,
-					Name:      configVolume,
-				}},
-				Ports: ports,
-				LivenessProbe: &corev1.Probe{
-					ProbeHandler:   healthProbe,
-					TimeoutSeconds: healthTimeoutSeconds,
-					PeriodSeconds:  livenessPeriodSeconds,
-				},
-				StartupProbe: &corev1.Probe{
-					ProbeHandler:     healthProbe,
-					TimeoutSeconds:   healthTimeoutSeconds,
-					PeriodSeconds:    startupPeriodSeconds,
-					FailureThreshold: startupFailureThreshold,
-				},
-			}},
-			ServiceAccountName: constants.GoflowKubeName,
+			Containers:         []corev1.Container{container},
+			ServiceAccountName: constants.FLPName,
 		},
 	}
 }
 
-func buildMainCommand(desired *flowsv1alpha1.FlowCollectorGoflowKube) string {
-	return fmt.Sprintf(`/goflow-kube -loglevel "%s" -config %s/%s -healthport %d`,
-		desired.LogLevel, configPath, configFile, desired.HealthPort)
-}
-
 // returns a configmap with a digest of its configuration contents, which will be used to
 // detect any configuration change
 func (b *builder) configMap() (*corev1.ConfigMap, string) {
 	configStr := `{}`
-	config := &ConfigMap{
-		Listen:      fmt.Sprintf("netflow://:%d", b.desired.Port),
-		Loki:        LokiConfigMap{},
-		PrintInput:  false,
-		PrintOutput: b.desired.PrintOutput,
+	config := &Config{
+		LogLevel: b.desired.LogLevel,
+		Health:   Health{Port: int(b.desired.HealthPort)},
+		Pipeline: Pipeline{
+			Ingest: Ingest{Type: "collector", Collector: Collector{
+				Hostname: "0.0.0.0",
+				Port:     int(b.desired.Port),
+			}},
+			Decode:  Decode{Type: "json"},
+			Encode:  Encode{Type: "none"},
+			Extract: Extract{Type: "none"},
+			Transform: []Transform{
+				{Type: "network", Network: Network{
+					Rules: []Rule{
+						{
+							Input:  "SrcAddr",
+							Output: "SrcK8S",
+							Type:   "add_kubernetes",
+						},
+						{
+							Input:  "DstAddr",
+							Output: "DstK8S",
+							Type:   "add_kubernetes",
+						},
+					},
+				}}},
+			Write: Write{Type: "loki", Loki: Loki{Labels: constants.LokiIndexFields}},
```

Comment:
Should we expose `decode` `transform` & `write` in `FlowCollector` config if you simply want to collect and send to something else without using the console UI ?

Reply:
Yes that could be interesting. For now I prefer to focus on FLP integration with as few changes as possible in the current API, but we can expect more config exposition soon. We need also to draw the line between something opinionated or not. FLP is flexible, but the operator could be more opinionated (which means easier to configure). Not saying it has to be.. it's a discussion to have